src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1489,7 +1489,9 @@ def _load_checkpoint_to_model(self):  # noqa: C901 - too complex
 
         if self.load_checkpoint or self.external_checkpoint_path:
             # GET LOCAL PATH TO THE CHECKPOINT FILE FIRST
+            ckpt_root_dir = str(Path(self.checkpoints_dir_path).parent)
             ckpt_local_path = get_ckpt_local_path(
+                ckpt_root_dir=ckpt_root_dir,
                 source_ckpt_folder_name=self.source_ckpt_folder_name,
```

Comment:
Not directly related, but it looks like `self.source_ckpt_folder_name` is always None int he Trainer. I think it would be better to remove because it makes it confusing. Typically, I first wondered what is the difference between `ckpt_root_dir` vs `self.source_ckpt_folder_name` ?

Reply:
removed (including from weight averaging) in latest commit.